toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,222 @@
+	initramfsPath := filepath.Join("boot", "initramfs-*")
+
+	// Fetch the initramfs file name.
+	var initramfsFiles []string
+	err := imageChroot.Run(func() error {
+		var innerErr error
+		initramfsFiles, innerErr = filepath.Glob(initramfsPath)
+		return innerErr
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list initramfs file: %w", err)
+	}
+
+	// Ensure an initramfs file is found
+	if len(initramfsFiles) != 1 {
+		return fmt.Errorf("expected one initramfs file, but found %d", len(initramfsFiles))
+	}
+
+	newInitramfs := filepath.Base(initramfsFiles[0])
+
+	cfgPath := filepath.Join("boot", "mariner.cfg")
+
+	// Update mariner.cfg to reference the new initramfs
+	err = imageChroot.Run(func() error {
+		input, innerErr := ioutil.ReadFile(cfgPath)
+		if innerErr != nil {
+			return fmt.Errorf("failed to read mariner.cfg: %w", innerErr)
+		}
+
+		lines := strings.Split(string(input), "\n")
+		for i, line := range lines {
+			if strings.HasPrefix(line, "mariner_initrd=") {
+				lines[i] = "mariner_initrd=" + newInitramfs
+			}
+		}
+		output := strings.Join(lines, "\n")
+		return ioutil.WriteFile(cfgPath, []byte(output), 0644)
+	})
+
+	return nil
+}
+
+func updateGrubConfig(resolvedVerityDevice string, resolvedHashDevice string, salt string, rootHash string, verityCorruptionResponse string) error {
+	const cmdlineTemplate = "rd.systemd.verity=1 roothash=%s systemd.verity_root_data=%s systemd.verity_root_hash=%s systemd.verity_root_options=%s,salt=%s"
+	newArgs := fmt.Sprintf(cmdlineTemplate, rootHash, resolvedVerityDevice, resolvedHashDevice, verityCorruptionResponse, salt)
+	grubConfigPath := "/mnt/boot_partition/grub2/grub.cfg"
+
+	content, err := ioutil.ReadFile(grubConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to read grub config: %v", err)
+	}
+
+	// Split the content into lines for processing
+	lines := strings.Split(string(content), "\n")
+	var updatedLines []string
+
+	for _, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+		if strings.HasPrefix(trimmedLine, "linux ") {
+			// Append new arguments to the line that starts with "linux"
+			line += " " + newArgs
+		}
+		if strings.HasPrefix(trimmedLine, "set rootdevice=PARTUUID=") {
+			// Replace the root device line with the new root device. TODO: add supported type 'user'
+			line = "set rootdevice=/dev/mapper/root"
+		}
+		updatedLines = append(updatedLines, line)
+	}
+
+	// Write the updated content back to grub.cfg
+	err = ioutil.WriteFile(grubConfigPath, []byte(strings.Join(updatedLines, "\n")), 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write updated grub config: %v", err)
+	}
+
+	return nil
+}
+
+// findFreeNBDDevice finds the first available NBD device.
+func findFreeNBDDevice() (string, error) {
+	files, err := filepath.Glob("/sys/class/block/nbd*")
+	if err != nil {
+		return "", err
+	}
+
+	for _, file := range files {
+		size, err := ioutil.ReadFile(filepath.Join(file, "size"))
```

Comment:
```suggestion
		size, err := os.ReadFile(filepath.Join(file, "size"))
```
`ioutil` is deprecated